### PR TITLE
Add ability to cancel render for specific node, update readme

### PR DESCRIPTION
**Pull request by likerRr**
#109 Attempt to add support for cancelling single timer for specific node

minified size increased to 2.02 kb :disappointed: If someone has better ideas - lets discuss them together

Also this PR fixes unpleasant issue with this [code](https://github.com/hustcc/timeago.js/blob/master/src/timeago.js#L146) - there are a lot of "dead" timers that are stored in the array until they are explicity removed by calling `cancel`. By "dead" timers I mean timers that are already executed but their ids are still in the array. Potentially it causes memory leaks.

### CODE REVIEW

**Comment by likerRr** [comment]
Since `cancel` method doesn't relate to a Timeago instance, it was moved out as a static property

**Comment by hustcc** [comment]
Maybe code below can save file bit:

```js
timer[node] = setTimeout(functon, timeout)...
```

**Comment by likerRr** [comment]
Object can't be a key name of another object. Only strings/numbers/booleans are allowed. If only we can use es6, then there is very good structure - Map, which is allow objects to be keys

**Comment by hustcc** [comment]
:haircut_woman: just tested in chrome, it is OK. ^_^ sorry.

In addition, maybe we should notice the api compatibility of `cancel`.

File: src/timeago.js [hunk 1]
```diff
@@ -35,14 +35,9 @@ function () {
...
 
   function findTimeoutByNode(node) {
-    var result, key;
-    for (key in timers) {
-      if (timers[key] === node) {
-        result = key;
-        break;
-      }
+    for (var key in timers) {
+      if (timers[key] === node) return key;
```
Comment (likerRr) [inline src/timeago.js]: Shame on me :smile:

Comment (hustcc) [reply src/timeago.js]: :face_with_thermometer:

**Comment by likerRr** [comment]
If we accept this PR, then by following [semver](http://semver.org/) the major version should be incremented, because as you noticed PR introduces incompatible API change.

**Comment by hustcc** [comment]
So version 3.0.x is coming. ^_^

**Comment by likerRr** [comment]
@hustcc it would be nice release: 
1. typescript support
2. ability to cancel single timer
3. performance improvement by utilizing prototypal inheritance

:+1:

**Comment by likerRr** [comment]
dist folder is not updated

**Comment by hustcc** [comment]
So timeago.d.ts should be:

```ts
/// <reference types="jquery" />
type TDate = Date | string | number;
export interface TimeAgoConstructor {
    new ();
    new (nowDate: TDate, defaultLocale?: string);
    format(date: TDate, locale?: string): string;
    render<T>(nodes: Node | NodeList | JQuery, locale?: string): void;
    setLocale(locale: string): void;
}
export interface TimeagoFactory {
    (): TimeAgoConstructor;
    (nowDate: TDate, defaultLocale?: string): TimeAgoConstructor;
    cancel(node?: Node | JQuery): void;
    register(locale: string, localeFunc: Function): void;
}
declare let timeagoFactory: TimeagoFactory;
export default timeagoFactory;
```

I have never wrote ts.

**Comment by likerRr** [comment]
You absolutely right: `cancel` moves to factory as a static method

**Comment by hustcc** [comment]
`?:` means not required paramter ?

**Comment by likerRr** [comment]
Correct, it means optional parameter + `null` value

**Pull request merged**